web3/contract.py

```diff
@@ -1323,6 +1350,72 @@ def build_filter(self):
         builder.address = self.address
         return builder
 
+    @combomethod
+    def getLogs(self,
+                argument_filters=None,
+                fromBlock=1,
+                toBlock="latest"):
+        """Get events for this contract instance using eth_getLogs API.
+
+        This is a stateless method, as opposite to createFilter.
+        It can be safely called against nodes which do not provide
+        eth_newFilter API, like Infura nodes.
+
+        If no block range is provided and there are many events,
+        like ``Transfer`` events for a popular token,
+        the Ethereum node might be overload and timeout
+        on underlying JSON-RPC call.
+
+        Example - how to get all ERC-20 token transactions
+        for the latest 10 blocks:
+
+        .. code-block:: python
+
+            f = max(mycontract.web3.eth.blockNumber - 10, 1)
+            t= mycontract.web3.eth.blockNumber
+
+            events = mycontract.events.Transfer.getLogs(fromBlock=f, toBlock=t)
+
+            for e in events:
+                print(e["args"]["from"],
+                    e["args"]["to"],
+                    e["args"]["value"])
+
+        :param argument_filters: TODO
+        :param fromBlock: block number, defaults to 1
+        :param toBlock: "block number or "latest", defaults to "latest"
+        :yield: Iterable of dictionatries
+        """
+
+        if not self.address:
+            raise TypeError("This method can be only called on "
+                "an instiated contract with an adress")
+
+        abi = self._get_event_abi()
+
+        if argument_filters is None:
+            argument_filters = dict()
+
+        _filters = dict(**argument_filters)
+
+        # Construct JSON-RPC raw filter presentation based on human readable Python descriptions
+        # Namely, convert event names to their keccak signatures
+        data_filter_set, event_filter_params = construct_event_filter_params(
+            abi,
+            contract_address=self.address,
+            argument_filters=_filters,
+            fromBlock=fromBlock,
+            toBlock=toBlock,
+            address=self.address,
+        )
+
+        # Call JSON-RPC API
+        logs = self.web3.eth.getLogs(event_filter_params)
+
+        # Convert raw binary data to Python proxy objects as described by ABI
+        for entry in logs:
+            yield get_event_data(abi, entry)
```

Comment:
I think using a `yield` pattern here is likely to cause confusion.  Maybe go ahead and decorate this function using `to_tuple` or turn this into  `return tuple(get_event_data(abi, entry) for entry in logs)`

Reply:
Alrighty. We are not streaming the response from the node any case, so there is little marginal gain here.

Reply:
Fixed in the latest revision.